src/client/api/event.c

```diff
@@ -525,7 +525,8 @@ ev_progress_cb(void *arg)
 	/** Change status of event to INIT only if event is not in EQ and get out. */
 	if (daos_handle_is_inval(evx->evx_eqh)) {
 		D_MUTEX_LOCK(&evx->evx_lock);
-		evx->evx_status = DAOS_EVS_READY;
+		if (evx->evx_status == DAOS_EVS_COMPLETED || evx->evx_status == DAOS_EVS_ABORTED)
+			evx->evx_status = DAOS_EVS_READY;
```

Comment:
This surely seems to be an improvement; I wonder about its sufficiency though.

1. If the event is neither COMPLETED nor ABORTED after we acquire the lock, should we still return 1?
2. How can the status of the event change after the lockless checks? This makes me nervous about the lockless checks.

Reply:
Oops, should have started the review on the master pull request; I'll copy the comment over.